### ISSUE TITLE
XML feedy: Ignorovat produkty bez zadaných údajů

### ISSUE THREAD

**Issue by pavelevap** [pavelevap/ceske-sluzby#76]
Cena je např. pro Heureku důležitý a povinný parametr, ale někdy nemusí být u produktu uvedena. Je potřeba projít jednotlivé parametry a zkontrolovat, aby se do feedu nedostalo nic, co by mohlo způsobit problémy.

**Comment by cJayyy** [comment]
Já navrhuji udělat tuto funkci jako možnost. Uživatel si vybere, jestli chce, nebo nechce ve feedu vynechat produkty bez ceny.

**Comment by cJayyy** [comment]
Je v tomto problému nějaký progres?

**Comment by pavelevap** [comment]
@cJayyy: Bohužel zatím nikoli, dodělával jsem nejdříve Google feed a nastavení dodací doby :-( Ale už se k tomu snad dostanu, nemělo by to být nic složitého, pouze ohlídat některé informace...

Nemáte nějaký přehled, co by se mělo všechno kontrolovat? Aneb co je u různých feedů povinné?

**Comment by cJayyy** [comment]
Také si myslím, že by to nemělo být nic složitého. Určitě by se to hodilo nejen mě.

Přehled povinných údajů ve feedech nemám, ale snad všechny servery je mají k dispozici.

> Heuréka: http://sluzby.heureka.cz/napoveda/xml-feed/
> Zbozi: http://napoveda.seznam.cz/cz/zbozi/specifikace-xml-pro-obchody/specifikace-xml-feedu/

Víc jich nepoužívám, takže nevím.

**Comment by pavelevap** [comment]
Odkazy na dokumentaci znám také :-)

Pro **Zboží.cz** jsou tam jako povinné označené následující položky:

`PRODUCTNAME`
`DESCRIPTION`
`URL` - Adresu a `ID` musí mít každý produkt, to nemusíme asi řešit.
`PRICE_VAT`
`DELIVERY_DATE`

Pokud jde o **Heureku**, tak tam nikde nevidím, co je opravdu povinné? Jsou tam označené důležité elementy:

`PRODUCTNAME`
`ITEM_ID`
`CATEGORYTEXT` - Každý produkt musí být alespoň v nějaké kategorii, takže to asi není zásadní problém.
`DELIVERY_DATE`

Ale nevím, máte s tím někdo zkušenosti? Nechtěl bych to zakazovat, když to třeba není potřeba.

**Comment by cJayyy** [comment]
Toto je e-mail od Heureky:

> Pro každý produkt musí být uvedena cena. Zkontrolujte, prosím, zda u některých produktů nevynecháváte informaci o ceně, neuvádíte prázdný tag <PRICE_VAT> (příp. <PRICE>), nebo nulovou cenu.
> Pokud se jedná o neprodejné produkty (jako vzorníky, nové produkty s nenadefinovanou cenou, atp.) nesmí být uvedeny v XML feedu. V XML feedu je potřeba uvádět pouze produkty, které lze ve Vašem obchodě samostatně zakoupit.

Můj problém spočívá v tom, že na e-shopu mám produkty bez ceny, které jsou v nabídce ale pro jejich zakoupení zákazník musí napsat/zavolat. Celkem chápu, že takovéhle produkty nechce Heureka importovat. Já tedy potřebuji mít možnost některé produkty nezahrnovat do feedu.

**Comment by pavelevap** [comment]
@cJayyy: Ano, tomu rozumím. Automaticky je vynechávat ještě neumíme (pokud třeba nemají uvedenou cenu), ale doplním to... Každopádně ručně už můžete odebrat celé kategorie nebo jen jednotlivé produkty: #2. Stačí zaškrtnout v nastavení kategorie nebo produktu.

**Comment by cJayyy** [comment]
Super, to je zatím přijatelný způsob (v menším měřítku). Děkuji :)
